manifests/install.pp

```diff
@@ -6,4 +6,48 @@
   package { 'falco':
     ensure => $falco::package_ensure,
   }
+
+  # Install driver dependencies
+  # Dependencies are not required for modern-bpf driver
+  unless $falco::driver == 'modern-bpf' {
+    $_suse_kernel_version_sans_default = regsubst($facts['kernelrelease'], '^(.*)-default$', '\\1')
+    $_running_kernel_devel_package = $facts['os']['family'] ? {
+      'Debian' => "linux-headers-${facts['kernelrelease']}",
+      'RedHat' => "kernel-devel-${facts['kernelrelease']}",
+      'Suse'   => "kernel-default-devel-${_suse_kernel_version_sans_default}",
+      default  => fail("The module \"${module_name}\" does not yet support \"${facts['os']['family']}\""),
+    }
+    ensure_packages([$_running_kernel_devel_package], { 'before' => Package['falco'] })
+
+    $_package_deps = ['dkms', 'make']
+    ensure_packages($_package_deps, { 'before' => Package['falco'] })
+
+    $_driver_type = $falco::driver ? {
+      'kmod'  => 'module',
+      'bpf'   => 'bpf',
+      default => fail("The drvier \"${falco::driver}\" is not yet supported by either the module \"${module_name}\" or \"falco-driver-loader\""), # lint:ignore:140chars
+    }
+
+    # Download and compile the desired falco driver based on the currently running kernel version.
+    # Recompile if the running kernel version change or falco package changes.
+    #
+    # Note, the default "--compile" flag should not be needed, but there appears to be a bug.
+    # Open issue at https://github.com/falcosecurity/falco/issues/2431
+    $_kernel_mod_path = $facts['os']['family'] ? {
+      'Debian' => "/lib/modules/${facts['kernelrelease']}/updates/dkms/falco.ko",
+      'RedHat' => "/lib/modules/${facts['kernelrelease']}/extra/falco.ko.xz",
+      'Suse'   => "/lib/modules/${facts['kernelrelease']}/updates/falco.ko",
+      default  => fail("The module \"${module_name}\" does not yet support \"${facts['os']['family']}\""),
+    }
+
+    exec { "falco-driver-loader ${_driver_type} --compile":
+      creates   => $_kernel_mod_path,
+      path      => '/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin',
+      subscribe => [
+        Package[$_running_kernel_devel_package],
+        Package['falco'],
```

Comment:
```suggestion
        Package[
          $_running_kernel_devel_package,
          'falco',
        ],
```

Reply:
I was leaning towards not just because, even though there's some significant underlying changes, this technically didn't change any default behavior or signatures to the way the class is used, other than requiring a minimum package version. Internally, we didn't have to add/modify/remove any class parameters.

Happy to add a note if you think it's worthwhile though.